qutip/wavefunction/wavefunction_cython.pyx

```diff
@@ -0,0 +1,456 @@
+
+    Examples
+    --------
+    ```python
+    >>> psi_n_single_fock_single_position_complex(0,1.0+2.0j)
+    (-1.4008797330262455-3.0609780602975003j)
+    >>> psi_n_single_fock_single_position_complex(61,1.0+2.0j)
+    (-511062135.47555304+131445997.75753704j)
+    ```
+
+    References
+    ----------
+    - Pérez-Jordá, J. M. (2017). On the recursive solution of the quantum harmonic oscillator. *European Journal of Physics*, 39(1), 
+      015402. doi:10.1088/1361-6404/aa9584
+    """
+    
+    cdef np.ndarray[np.float64_t, ndim=1] n_coeffs
+
+    r0 = 0.0 + 0.0j
+    r1 = (cpi ** (-0.25)) * cexp(-(x ** 2) / 2)
+
+    for index in range(n):
+        if index == 0:
+            r2 = 2 * x * (r1 / csqrt(2 * (index + 1)))
+            r0 = r1
+            r1 = r2
+        else:
+            r2 = 2 * x * (r1 / csqrt(2 * (index + 1))) - csqrt(index / (index + 1)) * r0 
+            r0 = r1
+            r1 = r2
+
+    return r1
+
+@cython.nogil
+@cython.cfunc
+@cython.locals(x_size=np.npy_intp, j=int, i=int, k=int, temp1=double, temp2=double)
+@cython.boundscheck(False)
+cpdef np.ndarray[np.float64_t, ndim=1] psi_n_single_fock_multiple_position(int n, np.ndarray[np.float64_t, ndim=1] x):
+
+    """
+    Compute the wavefunction to a real vector x using adapted recurrence relation.
+
+    Parameters
+    ----------
+    n : int
+        Quantum state number.
+    x : np.ndarray[np.float64_t]
+        Position(s) at which to evaluate the wavefunction.
+
+
+    Returns
+    -------
+        np.ndarray[np.float64_t]
+        The evaluated wavefunction.
+
+    Examples
+    --------
+    ```python
+    >>> psi_n_single_fock_multiple_position(0, np.array([1.0, 2.0]))
+    array([0.45558067, 0.10165379])
+    >>> psi_n_single_fock_multiple_position(61, np.array([1.0, 2.0]))
+    array([-0.23930492, -0.01677378])
+    ```
+
+    References
+    ----------
+    - Pérez-Jordá, J. M. (2017). On the recursive solution of the quantum harmonic oscillator. *European Journal of Physics*, 39(1), 
+      015402. doi:10.1088/1361-6404/aa9584
+    """
+    
+    x_size = x.shape[0]
+    cdef np.ndarray[np.float64_t, ndim=2] result = np.zeros((n + 1, x_size), dtype=np.float64)
+
+    for j in range(x_size):
+        result[0, j] = (pi ** (-0.25)) * exp(-(x[j] ** 2) / 2)
+
+    for i in range(n):
+        temp1 = sqrt(2 * (i + 1))
+        temp2 = sqrt(i / (i + 1))
+        if(i == 0):
+            for k in range(x_size):
+                result[i + 1, k] = 2 * x[k] * (result[i, k] / temp1)
+        else:
+            for k in range(x_size):
+                result[i + 1, k] = 2 * x[k] * (result[i, k] / temp1) - temp2 * result[i - 1, k]
+
+    return result[-1, :]
+
+@cython.nogil
+@cython.cfunc
+@cython.locals(x_size=np.npy_intp, j=int, i=int, k=int, temp1=complex, temp2=complex)
+@cython.boundscheck(False)
+cpdef np.ndarray[np.complex128_t, ndim=1] psi_n_single_fock_multiple_position_complex(int n, np.ndarray[np.complex128_t, ndim=1] x):
```

Comment:
This is the only function used...
Can you remove the others?

Reply:
I solved it